modules/generative-cohere/clients/cohere.go

```diff
@@ -189,22 +190,23 @@ func (v *cohere) getApiKey(ctx context.Context) (string, error) {
 }
 
 type generateInput struct {
-	Prompt            string   `json:"prompt"`
-	Model             string   `json:"model"`
-	MaxTokens         int      `json:"max_tokens"`
-	Temperature       int      `json:"temperature"`
-	K                 int      `json:"k"`
-	StopSequences     []string `json:"stop_sequences"`
-	ReturnLikelihoods string   `json:"return_likelihoods"`
+	ChatHistory   []message `json:"chat_history"`
+	Message       string    `json:"message"`
+	Model         string    `json:"model"`
+	MaxTokens     int       `json:"max_tokens"`
+	Temperature   int       `json:"temperature"`
+	K             int       `json:"k"`
+	StopSequences []string  `json:"stop_sequences"`
 }
 
-type generateResponse struct {
-	Generations []generation
-	Error       *cohereApiError `json:"error,omitempty"`
```

Comment:
Please don't remove the error, look [here](https://docs.cohere.com/reference/errors)